### ISSUE TITLE
GcStress: Linux: Sanity check failure in _il_dbggcval_nested.exe

### ISSUE THREAD

**Issue by swaroop-sridhar** [dotnet/coreclr#2769]
```
corerun JIT/Methodical/tailcall/_il_dbggcval_nested/_il_dbggcval_nested.exe

Assert failure(PID 22057 [0x00005629], Thread: 22057 [0x5629]): SanityCheck()
    File: /home/swaroops/git/a/coreclr/src/vm/methodtable.cpp Line: 9739
    Image: /home/swaroops/git/a/coreclr/bin/Product/Linux.x64.Debug/corerun

**** MessageBox invoked, title 'corerun - Assert Failure (PID 22057, Thread 22057/0x5629)' ****
  SanityCheck()

/home/swaroops/git/a/coreclr/src/vm/methodtable.cpp, Line: 9739

Abort - Kill program
Retry - Debug
Ignore - Keep running


Image:
/home/swaroops/git/a/coreclr/bin/Product/Linux.x64.Debug/corerun

********

/home/swaroops/bin/runn: line 3: 22057 Aborted                 (core dumped) 
```

Repro steps are:

* cp \scratch2\scratch\swaroops\coredistools\libcoredistools.so somewhere
* export LD_LIBRARY_PATH=somewhere
* export COMPlus_GCStress=0xC
* Run the above test

**Comment by sivarv** [comment]
This is a case of methods returning a struct with two gc-ref fields.  This must be failing for the same reason as issue #2757

**Comment by swaroop-sridhar** [comment]
This failure in _speed_relcall with JitStress seems to be the same failure.
@LLITCHEV: Can you please confirm? Thanks.

```
corerun JIT/Methodical/VT/callconv/_speed_relcall/_speed_relcall.exe
ISomething::DoSomething - 0/- 10/10
this is {10/10}
ISomething::RetSomething - 10/10
this is {10/10}
Got : 10/10
ISomething::DoSomething - 0/- 10/10
this is {10/10}
ISomething::RetSomething - 10/10
this is {10/10}
Got : 10/10
ISomething::DoSomething - 0/- 11/11
this is {10/10}
ISomething::RetSomething - 11/11
this is {10/10}

Assert failure(PID 2951 [0x00000b87], Thread: 2951 [0x0b87]): SanityCheck()
    File: /home/swaroops/git/coreclr/src/vm/methodtable.cpp Line: 9738
    Image: /home/swaroops/git/coreclr/bin/Product/Linux.x64.Debug/corerun

**** MessageBox invoked, title 'corerun - Assert Failure (PID 2951, Thread 2951/0x0b87)' ****
  SanityCheck()

/home/swaroops/git/coreclr/src/vm/methodtable.cpp, Line: 9738
Abort - Kill program
Retry - Debug
Ignore - Keep running

Image:
/home/swaroops/git/coreclr/bin/Product/Linux.x64.Debug/corerun
********
/home/swaroops/bin/runn: line 3:  2951 Aborted                 (core dumped) ./corerun ~/test/$1.exe
134
```

**Comment by LLITCHEV** [comment]
These two bugs seems to have been fixed with PR #2860
